genshin/client/components/hoyolab.py

```diff
@@ -29,6 +29,25 @@ async def search_users(
         )
         return [models.PartialHoyolabUser(**i["user"]) for i in data["list"]]
 
+    async def get_hoyolab_self(self, *, lang: typing.Optional[str] = None) -> models.PartialHoyolabUser:
```

Comment:
Please combine this with `get_hoyolab_user`, no need for a new function.

Reply:
The problem is , get_hoyolab_user returns FullHoyolabUser, that works fine for oversea cookies and throws bunch of errors on cn cookies. json returned aren't actually compatible.

Reply:
Unless get_hoyolab_user  returns PartialHoyolabUser too

Reply:
Fixed.